### PR TITLE
Build improvement closes #36

### LINKED ISSUE #36: Makefile improvements
I think that this issue can be implemented in multiple PRs,

## Echoing the command output is useful

Every single command in the makefile is not echoing the command output, it's very annoying as one of the most important features of Make is that it shows you what it's doing.

## Dockerfile not aligned with Makefile

There are actually two dockerfiles, named `Dockerfile.image` and `Dockerfile.build` .
They can me merged in a single multi staged build Dockerfile and also the image dockerfile has the orbiter command both in `entrypoint` and in `cmd` and this does not work. (no review on that?) Introduced in https://github.com/gianarb/orbiter/pull/30

## PHONY

For example, the `binaries` target should just be an alias to a `bin/orbiter` target without PHONY
Instead `build/docker` should be a PHONY target

### PR DESCRIPTION
**Pull request by fntlnz**
- Some cleaning on the Makefile, also specify PHONYs
- Static builds
- Use multi staged builds for the Docker image
- Remove .idea from the gitignore, it's not related to the project but on
how the user interacts with it. If one uses idea he can just create a
global gitignore to ignore that
- License had a disalignement

Signed-off-by: Lorenzo Fontana <lo@linux.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,11 @@ services:
   - docker
 script:
   #- sudo PATH=$PATH GOPATH=$GOPATH make ci binaries
-  - make ci binaries
+  - make ci
 after_success:
-  - docker login -u $DOCKER_USER -p $DOCKER_PASS
+  - if [ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_PULL_REQUEST_BRANCH" == "" ]; docker login -u $DOCKER_USER -p $DOCKER_PASS
```
Comment (fntlnz) [inline .travis.yml]: This is better to be implemented with an external script

File: .travis.yml [hunk 2]
```diff
@@ -5,11 +5,11 @@ services:
   - docker
 script:
   #- sudo PATH=$PATH GOPATH=$GOPATH make ci binaries
```
Comment (fntlnz) [inline .travis.yml]: @gianarb can be removed?

File: .travis.yml [hunk 3]
```diff
@@ -5,11 +5,11 @@ services:
...
-  - make ci binaries
+  - make ci
```
Comment (fntlnz) [inline .travis.yml]: Please note that I'm not compiling, just doing the CI here. If you need to compile the `build` target can be added as dep to the CI target

**Comment by gianarb** [comment]
Thanks a lot!

File: .travis.yml (hunk 1 above)
Comment (gianarb) [inline .travis.yml]: You can do that in another PR! I merge this now.

**Comment by gianarb** [comment]
Do you think we should maintain other repository as quay?
